electrumx/server/http_middleware.py

```diff
@@ -43,29 +33,70 @@ async def check_limit(self):
             return True
         return False
 
+
 rate_limiter = RateLimiter(
     max_tokens=os.environ.get("MAX_TOKENS", 10000),
     refill_interval=os.environ.get("RATE_LIMIT_WINDOW_SECONDS", 3),
     delay_after=os.environ.get("RATE_LIMIT_DELAY_AFTER", 5),
     delay_ms=os.environ.get("RATE_LIMIT_DELAY_MS", 300)
 )
 
+
+async def cors_middleware() -> web_middlewares:
+    async def factory(handler):
```

Comment:
And all the factory define should be `async def factory(app: web.Application, handler):` factory method need to args, `app` and `handler`